packages/@aws-cdk/aws-glue/lib/table.ts

```diff
@@ -287,6 +328,80 @@ export class Table extends Resource implements ITable {
       resourceName: `${this.database.databaseName}/${this.tableName}`,
     });
     this.node.defaultChild = tableResource;
+
+    // Partition index creation relies on created table.
+    if (props.partitionIndexes) {
+      if (props.partitionIndexes.length > 3) {
+        throw new Error(`Maximum number of partition indexes allowed is 3 but got ${props.partitionIndexes.length}`);
+      }
+      this.partitionIndexes = props.partitionIndexes;
+      this.partitionIndexes.forEach((index) => this.addPartitionIndex(index));
+    }
+  }
+
+  /**
+   * Add a partition index to the table. You can have a maximum of 3 partition
+   * indexes to a table. Partition index keys must be a subset of the table's
+   * partition keys.
+   *
+   * @see https://docs.aws.amazon.com/glue/latest/dg/partition-indexes.html
+   */
+  public addPartitionIndex(index: PartitionIndex) {
+    const numPartitions = this.partitionIndexCustomResources.length;
+    if (numPartitions > 3) {
```

Comment:
⚠️ Off-by-one error. If the length is already 3, we're attempting to add a 4th.

```suggestion
    if (numPartitions >= 3) {
```

If does seem like you could just keep this check, and remove the one in the constructor to avoid the duplication.

Reply:
good catch, I moved the logic up from the bottom of the function and didn't think through the implications